examples/matmul.mojo

```diff
@@ -202,31 +205,33 @@ fn matmul_tiled_unrolled_parallelized(C: Matrix, A: Matrix, B: Matrix):
         alias tile_size = 4
         tile[calc_tile, nelts * tile_size, tile_size](A.cols, C.cols)
 
-    parallelize[calc_row](C.rows)
+    parallelize[calc_row](rt, C.rows)
 
 
 @always_inline
 fn benchmark[
-    func: fn (Matrix, Matrix, Matrix) -> None
+    func: fn (Matrix, Matrix, Matrix, Runtime) -> None
 ](M: Int, N: Int, K: Int, base_gflops: Float64, str: String):
     var C = Matrix(M, N)
     C.zero()
     var A = Matrix(M, K)
     var B = Matrix(K, N)
 
-    @always_inline
-    @parameter
-    fn test_fn():
-        _ = func(C, A, B)
-
-    let secs = Float64(Benchmark().run[test_fn]()) / 1_000_000_000
-    # Prevent the matrices from being freed before the benchmark run
-    _ = (A, B, C)
-    let gflops = ((2 * M * N * K) / secs) / 1e9
-    let speedup: Float64 = gflops / base_gflops
-    # print(gflops, "GFLOP/s", speedup, " speedup")
-    print(str)
-    print(gflops, "GFLOP/s <>", speedup.to_int(), "x speedup over Python")
+    with Runtime() as rt:
+
+        @always_inline
+        @parameter
+        fn test_fn():
+            _ = func(C, A, B, rt)
+
+        let secs = Float64(Benchmark().run[test_fn]()) / 1_000_000_000
+        # Prevent the matrices from being freed before the benchmark run
+        _ = (A, B, C)
+        let gflops = ((2 * M * N * K) / secs) / 1e9
+        let speedup: Float64 = gflops / base_gflops
+        # print(gflops, "GFLOP/s", speedup, " speedup")
+        print(str)
```

Comment:
Non-blocking suggestion, but at some point, we probably want the pretty print fn to reduce the precision. Perhaps in the next update:

```python
def float_printer(f: Float32):
    let builtin = Python.import_module('builtins')
    let pyf = builtin.float(f)
    builtin.print(builtin.str("{:4.2f}").format(pyf))
```

Reply:
Yes good idea, maybe we can even pop that in the std lib :)